packages/common/src/api/tan-query/useEmailInUse.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from '@tanstack/react-query'
+
+import { useAudiusQueryContext } from '~/audius-query'
+import { AudiusQueryContextType } from '~/audius-query/AudiusQueryContext'
+
+import { QUERY_KEYS } from './queryKeys'
+import { Config } from './types'
+
+export const fetchEmailInUse = async (
+  email: string | null | undefined,
+  { identityService }: AudiusQueryContextType
+) => {
+  if (!email) return { exists: false, isGuest: false }
+  return await identityService.checkIfEmailRegistered(email)
+}
+
+/**
+ * Hook to check if an email is already registered
+ */
+export const useEmailInUse = (
+  email: string | null | undefined,
+  config?: Config
+) => {
+  const context = useAudiusQueryContext()
+
+  return useQuery({
+    queryKey: [QUERY_KEYS.emailInUse, email],
+    queryFn: () => fetchEmailInUse(email, context),
+    staleTime: config?.staleTime,
+    enabled: config?.enabled !== false && !!email
```

Comment:
Nit on usage: The pattern of checking the arguments and conditionally setting `enabled` is fine. But I would love to get to a world where we just spread the config in here and rarely ever override it unless really needed.
We _could_ as an alternative, just return `false` immediately in the `queryFn` if we receive undefined/null and then that gets cached as the result for querying and we don't have to skip running the hook?

Just a thought, maybe it feels more confusing than using this pattern.